JavascriptSlideshowHooks.php

```diff
@@ -8,20 +11,20 @@
  * @link    https://gitlab.com/hydrawiki/extensions/javascriptslideshow
  */
 
-class JavascriptSlideshowHooks {
+class JavascriptSlideshowHooks implements ParserFirstCallInitHook {
 	/**
 	 * Sets up this extensions parser functions.
 	 *
-	 * @param Parser &$parser
+	 * @param Parser $parser
 	 * @return bool true
 	 * @throws MWException
 	 */
-	public static function onParserFirstCallInit( Parser &$parser ): bool {
+	public function onParserFirstCallInit( $parser ): bool {
 		$output = RequestContext::getMain()->getOutput();
 		$output->addModules( 'ext.slideshow.main' );
 
-		$parser->setHook( 'slideshow', [ __CLASS__, 'renderSlideshowTag' ] );
-		$parser->setFunctionHook( 'slideshow', [ __CLASS__, 'renderSlideshowParserFunction' ] );
+		$parser->setHook( 'slideshow', [ 'JavascriptSlideshowHooks', 'renderSlideshowTag' ] );
```

Comment:
`[ JavascriptSlideshowHooks::class, 'renderSlideshowTag' ]` Class name is better here instead of string literal if 
called methods are preferred to be kept static.

below purely optional:
Could be changed to `[ $this, 'renderSlideshowTag' ]` if there is incentive to rewrite methods here to be non-static ones.

Reply:
Pushed the change with class name